llvm/lib/SYCLLowerIR/ESIMD/LowerESIMDVLoadVStore.cpp

```diff
@@ -102,17 +102,15 @@ PreservedAnalyses ESIMDLowerLoadStorePass::run(Function &F,
       if (AS0 != AS1) {
         IRBuilder<> Builder(&Inst);
         if (GenXIntrinsic::isVStore(&Inst)) {
-          auto PtrTy = cast<PointerType>(Inst.getOperand(1)->getType());
-          PtrTy = PointerType::getWithSamePointeeType(PtrTy, AS1);
+          auto PtrTy = PointerType::get(Inst.getOperand(1)->getContext(), AS1);
```

Comment:
From the original code of getWithSamePointeeType, it seems like the new code should be 
auto PtrTy = PointerType::get(Inst.getOperand(1)->getType()->getContext(), AS1);
Can you please check?

Thanks

Reply:
I think they should be the same, but yeah, updated to getType()->getContext() for safe.